src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,179 @@
 import json
 import logging
+from typing import Dict, List
 
 import d3a.constants
-from d3a.d3a_core.exceptions import InvalidBidOfferPair
+from d3a.d3a_core.exceptions import (
+    InvalidBidOfferPairException, OfferNotFoundException,
+    BidNotFoundException, MycoValidationException)
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
-from d3a.models.market import validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import (
-    BidOfferMatch, offer_or_bid_from_json_string)
+from d3a.models.market import Market
+from d3a.models.market.market_structures import BidOfferMatch
 
 from d3a.models.myco_matcher.base_matcher import BaseMatcher
 
 
 class ExternalMatcher(BaseMatcher):
+    """Class responsible for external bids / offers matching."""
     def __init__(self):
         super().__init__()
         self.simulation_id = d3a.constants.COLLABORATION_ID
         self.myco_ext_conn = None
-        self.channel_prefix = f"external-myco/{self.simulation_id}/"
+        self.channel_prefix = f"external-myco/{self.simulation_id}"
+        self.response_channel = f"{self.channel_prefix}/response"
+        self.events_channel = f"{self.response_channel}/events/"
         self._setup_redis_connection()
         self.area_uuid_markets_mapping = {}
-        self.markets_mapping = {}
+        self.markets_mapping = {}  # Dict[market_id: market] mapping
         self.recommendations = []
 
     def _setup_redis_connection(self):
         self.myco_ext_conn = ResettableCommunicator()
         self.myco_ext_conn.sub_to_multiple_channels(
-            {"external-myco/get-simulation-id": self.get_simulation_id,
-             f"{self.channel_prefix}offers-bids/": self.publish_offers_bids,
-             f"{self.channel_prefix}post-recommendations/": self.match_recommendations})
+            {"external-myco/get-simulation-id": self.publish_simulation_id,
+             f"{self.channel_prefix}/offers-bids/": self.publish_offers_bids,
+             f"{self.channel_prefix}/post-recommendations/": self.match_recommendations})
 
     def publish_offers_bids(self, message):
         """Publish open offers and bids.
 
-        published data are of the following format
-        market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
+        published data are of the following format:
+            {"bids_offers": {`market_id` : {"bids": [], "offers": [] }, }}
         """
-        # TODO: message can contain filters
-        data = {"event": "offers_bids_response"}
+        response_data = {"event": "offers_bids_response"}
+        data = json.loads(message.get("data"))
+        filters = data.get("filters", {})
```

Comment:
If the `myco-api-client` sends a wrong request (e.g. it has a typo in the `"filters"` key) it won't receive any meaningful error notification. I think we should validate this data somehow on the external matcher side. Not necessarily in this PR, but I think it's quite important since the external matchers will be implemented from external agents.

Reply:
Also, I think it would be good to describe the structure of the expected `message` in the docstring.

Reply:
we are not enforcing filters atm and I see no point in doing this, this is why  I did not add any validation here.

Reply:
And the expected message structure is already described in the docstring,  it is not displayed by github as this was not changed, it was there since Test1 V2, anw I'll add the filters key.